### ISSUE TITLE
Bug in ```_validate_contraction_modes```: modes seem to be wrongfully assigned.

### ISSUE THREAD

**Issue by ax-le** [tensorly/tensorly#585]
#### Describe the bug
As far as I understand tensor contractions, it seems to me that ```modes1``` and ```modes2``` are wrongfully set in the function ```_validate_contraction_modes``` in [tensorly.tenalg.tenalg_utils.py](https://github.com/tensorly/tensorly/blob/main/tensorly/tenalg/tenalg_utils.py) when the number of ```modes``` is equal to 2.

#### Steps or Code to Reproduce

The bug appears in the following code block, when ```modes``` is of length 2 (e.g. modes = [0,1]).

```python
modes1, modes2 = modes
```

The code comes from line 22 of [tensorly.tenalg.tenalg_utils.py](https://github.com/tensorly/tensorly/blob/main/tensorly/tenalg/tenalg_utils.py).

#### Expected behavior

 ```modes1``` and ```modes2``` should both take the value of ```modes``` (in my example,  modes1 = [0,1] and modes2 = [0,1]).

#### Actual result

```modes1``` takes the first value and ```modes2``` the second one (in my example, modes1 = 0 and modes2 = 1).

### Potential solution

A potential solution should be easy to implement, by just deleting lines 21/22/23 (the try/except), and just letting lines 24 and 25 
```python
modes1 = modes
modes2 = modes
```

#### Versions
Linux-6.8.0-49-generic-x86_64-with-glibc2.39
Python 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0]
NumPy 2.0.2
SciPy 1.14.1
TensorLy 0.9.0

Thanks in advance! :)

**Comment by JeanKossaifi** [comment]
Hi @ax-le, thank you for the detailed context - I remember hesitating on this edge case when I wrote that function.

I chose the current behavior so that it defaults to a tensor contraction between mode1 and mode2: `tensordot(t1, t2, [m1, m2])` then contracts mode `m1` of `t1` with mode `m2` of tensor `t2`. Otherwise the syntax becomes very convoluted to do that :  `tensordot(t1, t2, [(m1, ), (m2, )])`

Either way if you got confused this clearly means the behavior should be better documented!

**Comment by ax-le** [comment]
Hi @JeanKossaifi thanks for your answer!

I understand your viewpoint, which was indeed very different from mine. I understand that, as it is, my pull request prevents your behaviour from being easily implemented, and I will hence probably retract it.

I still believe that there is a problem with the function as it is, because the current version of the code fires an error in my context - when using the contraction on both modes for both tensors (e.g. ```modes = [0,1]```). It seems to me that both our contexts are interfering one with another.

Do you think of a way for me to actually specify the function that I want contraction on both modes? Maybe  something like ```modes = [[0,1],[0,1]]``` ? 
I think that this use case should also be documented.

Thanks for the discussion!
And Happy Holidays! :)
ax-le.

**Comment by ax-le** [comment]
Hi @JeanKossaifi thanks for your answer!

I understand your viewpoint, which was indeed very different from mine. I understand that, as it is, my pull request prevents your behaviour from being easily implemented, and I will hence probably retract it.

I understand that I can achieve my behaviour - tensor contraction on both modes - by setting ```modes = [m1, m2]``` with ```m1=m2```. I agree that I was confused at first and that it would probably be better if documented!

Thanks for the discussion!
And Happy Holidays! :)
ax-le.

**Comment by JeanKossaifi** [comment]
Yes, indeed it should be better documented - would be great if you wanted to make a PR for that, otherwise I'll get to it as soon as I get a moment.

Thank you, and happy holidays to you as well @ax-le !

**Comment by ax-le** [comment]
I will not be able to do it before mid-January, I guess it can wait until there, except if you want to do it sooner! Let's keep us updated.

Thank you! :)
